pkg/disttask/framework/taskexecutor/task_executor.go

```diff
@@ -532,8 +533,13 @@ func (e *BaseTaskExecutor) onError(err error) {
 	if err == nil {
 		return
 	}
-	err = errors.Trace(err)
-	e.logger.Error("onError", zap.Error(err), zap.Stack("stack"))
+
+	if errors.HasStack(err) {
+		e.logger.Error("onError", zap.Error(err), zap.StackSkip("stack", 1), zap.String("error stack", fmt.Sprintf("%+v", err)))
```

Comment:
```suggestion
		e.logger.Error("onError", zap.Error(err), zap.Stack("stack"), zap.String("error stack", fmt.Sprintf("%+v", err)))
```

no need to change this. The top of stack can be `onError`. The issue is not related to "stack" field but "error stack" field. Also change below branch and remove the test for "stack"